pkg/schemadsl/generator/generator.go

```diff
@@ -91,7 +105,7 @@ func (sg *sourceGenerator) emitCaveat(caveat *core.CaveatDefinition) {
 
 		decoded, err := caveattypes.DecodeParameterType(caveat.ParameterTypes[paramName])
 		if err != nil {
-			panic("invalid parameter type on caveat")
+			return fmt.Errorf("invalid parameter type on caveat: %w", err)
```

Comment:
the assumption is we don'g use `MustBug` here because this is an input error

Reply:
Yep